test/error/bad_host_alignment.cpp

```diff
@@ -0,0 +1,35 @@
+#include "Halide.h"
+#include <stdio.h>
+
+using namespace Halide;
+using namespace Halide::Internal;
+
+IRPrinter irp(std::cerr);
+int main(int argc, char **argv) {
+    Func f;
+    Var x, y;
+    uint8_t a[10][10];
+
+    buffer_t buf;
+    buf.host = &a[0][0];
+    buf.extent[0] = 10;
+    buf.extent[1] = 10;
+    buf.stride[0] = 1;
+    buf.stride[1] = 10;
+    buf.elem_size = 1;
+    buf.min[0] = 0;
+    buf.min[1] = 0;
+    Buffer param_buf(UInt(8), &buf);
+    ImageParam in(UInt(8), 2);
+
+    in.set_host_alignment(512);
+    f(x, y) = in(x, y);
+    // f.compute_root();
```

Comment:
The error will be generated by a call to realize below. So this line would be executed, but it won't matter because it is the output func (and so is effectively compute_root by default).

Either way, it should be uncommented or removed.